File: scripts/send_enter_to_all.js
#!/usr/bin/env node

/**
 * Simple script to send Enter key to all tmux sessions to unstick any waiting for input
 * Refactored to use canonical TmuxInterface instead of duplicating tmux operations
 */

import { TmuxInterface } from '../src/tmux_interface.js';

async function main() {
    console.log('🔍 Looking for tmux sessions...');
    
    const tmux = new TmuxInterface();
    
    try {
        const sessions = await tmux.listSessions();
        
        if (sessions.length === 0) {
            console.log('No tmux sessions found.');
            return;
        }
        
        console.log(`\nFound ${sessions.length} tmux session(s):`);
        sessions.forEach(session => {
            console.log(`  - ${session.name}`);
        });
        
        console.log('\n📤 Sending Enter key to each session...');
        
        let successCount = 0;
        const execSessions = [];
        
        for (const session of sessions) {
            try {
                // Use canonical TmuxInterface sendKeys method
                const target = tmux.getPaneTarget(session.name);
                await tmux.sendKeys(target, '', true); // Send Enter key
                console.log(`✓ Sent Enter to: ${session.name}`);
                successCount++;
                
                // Track claude_exec_* sessions
                if (session.name.startsWith('claude_exec_')) {
                    execSessions.push(session.name);
                }
            } catch (error) {
                console.log(`✗ Failed to send Enter to: ${session.name} - ${error.message}`);
            }
        }
        
        console.log(`\n📊 Summary: Successfully sent Enter to ${successCount}/${sessions.length} sessions`);
        
        if (execSessions.length > 0) {
            console.log(`\n🎯 Found ${execSessions.length} claude_exec_* session(s) that were updated:`);
            execSessions.forEach(session => {
                console.log(`  - ${session}`);
            });
        }
        
        console.log('\n✅ Done!');
    } catch (error) {
        console.error('❌ Error:', error.message);
    }
}

// Run the script
main();

File: scripts/send_enter_to_all.py
#!/usr/bin/env python3
"""
Simple script to send Enter key to all tmux sessions to unstick any waiting for input
"""

import subprocess
import sys

def get_tmux_sessions():
    """Get list of all tmux sessions"""
    try:
        result = subprocess.run(['tmux', 'list-sessions'], 
                              capture_output=True, text=True, check=True)
        sessions = []
        for line in result.stdout.strip().split('\n'):
            if line:
                session_name = line.split(':')[0]
                sessions.append(session_name)
        return sessions
    except subprocess.CalledProcessError:
        print("No tmux server is running or no sessions found.")
        return []
    except Exception as e:
        print(f"Error getting sessions: {e}")
        return []

def send_enter_to_session(session_name):
    """Send Enter key to a specific session"""
    try:
        subprocess.run(['tmux', 'send-keys', '-t', session_name, 'Enter'], 
                      check=True)
        return True
    except subprocess.CalledProcessError as e:
        print(f"Failed to send Enter to {session_name}: {e}")
        return False

def main():
    print("🔍 Looking for tmux sessions...")
    
    sessions = get_tmux_sessions()
    
    if not sessions:
        print("No tmux sessions found.")
        return
    
    print(f"\nFound {len(sessions)} tmux session(s):")
    for session in sessions:
        print(f"  - {session}")
    
    print("\n📤 Sending Enter key to each session...")
    
    success_count = 0
    exec_sessions = []
    
    for session in sessions:
        if send_enter_to_session(session):
            print(f"✓ Sent Enter to: {session}")
            success_count += 1
            
            # Track claude_exec_* sessions
            if session.startswith('claude_exec_'):
                exec_sessions.append(session)
        else:
            print(f"✗ Failed to send Enter to: {session}")
    
    print(f"\n📊 Summary: Successfully sent Enter to {success_count}/{len(sessions)} sessions")
    
    if exec_sessions:
        print(f"\n🎯 Found {len(exec_sessions)} claude_exec_* session(s) that were updated:")
        for session in exec_sessions:
            print(f"  - {session}")
    
    print("\n✅ Done!")

if __name__ == "__main__":
    main()